content/content.go

```diff
@@ -72,31 +102,137 @@ func readFilesIntoFileContent(baseDir string, filelist []string) (map[string][]b
 	return filesContent, nil
 }
 
+// checkErrorKeysForMandatoryContent iterates over filenames defined in the mandatory files array; ensures all error keys have the attribute set
+func checkErrorKeysForMandatoryContent(errorKeys map[string]RuleErrorKeyContent) (valid bool) {
+	valid = true
+
+	for _, mandatoryFile := range MandatoryRuleWideContentFiles {
+		for errorKeyName, errorKey := range errorKeys {
+			// all error keys must have these attributes
+			switch mandatoryFile {
+			case GenericMarkdown:
+				if errorKey.Generic == "" {
+					log.Error().Msgf("Error key `%v` is missing mandatory file %v.", errorKeyName, GenericMarkdown)
+					valid = false
+					break
+				}
+			case ReasonMarkdown:
+				if errorKey.Reason == "" {
+					log.Error().Msgf("Error key `%v` is missing mandatory file %v.", errorKeyName, ReasonMarkdown)
+					valid = false
+					break
+				}
+			default:
+				log.Error().Msgf("Behaviour for mandatory file `%v` is not defined.", mandatoryFile)
+				valid = false
+				break
+			}
+		}
+	}
+
+	return
+}
+
+func copyContentToEmptyErrorKeys(
+	filename string,
+	ruleContent RuleContent,
+	errorKeys map[string]RuleErrorKeyContent,
+) {
+	for i, errorKey := range errorKeys {
+		ek := errorKey
+
+		switch filename {
+		case GenericMarkdown:
+			if errorKey.Generic == "" {
+				ek.Generic = ruleContent.Generic
+			}
+		case ReasonMarkdown:
+			if errorKey.Reason == "" {
+				ek.Reason = ruleContent.Reason
+			}
+		case SummaryMarkdown:
+			if errorKey.Summary == "" {
+				ek.Summary = ruleContent.Summary
+			}
+		case ResolutionMarkdown:
+			if errorKey.Resolution == "" {
+				ek.Resolution = ruleContent.Resolution
+			}
+		case MoreInfoMarkdown:
+			if errorKey.MoreInfo == "" {
+				ek.MoreInfo = ruleContent.MoreInfo
+			}
+		default:
+			log.Error().Msgf("Behaviour for copying contents of file `%v` to error keys is not defined.", filename)
+			break
+		}
+
+		errorKeys[i] = ek
+	}
+
+}
+
+func isPluginFileMandatory(fname string) bool {
+	for _, v := range RulePluginMandatoryContentFiles {
```

Comment:
dunno if `StringInSlice` from `utils` package will be useful there?

Reply:
ah, of course! thanks

Reply:
done